lib/utils.js

```diff
@@ -35,4 +35,42 @@ function isStream (stream) {
   return stream !== null && typeof stream === 'object' && typeof stream.pipe === 'function'
 }
 
-module.exports = { isGzip, isDeflate, isStream }
+// It is a helper used to provide a async iteratable for
+// Readable.from
+async function * intoAsyncIterator (payload) {
+  const isBuffer = Buffer.isBuffer(payload)
+
+  if (
+    (
+    // ArrayBuffer
+      payload instanceof ArrayBuffer ||
+    // NodeJS.TypedArray
+    ArrayBuffer.isView(payload)
+    ) &&
+    // Exclude Buffer to prevent double cast
+    !isBuffer
```

Comment:
first check if it is a buffer then do all the ArrayBuffer stuff.

Reply:
Having dinner right now, I will update later.

Reply:
Enjoy ;)